### PR TITLE
Dfk item transactions

### LINKED ISSUE #33: Table: DFK_Item_Transactions
```
with dfk_item_buy_sell as (
select tx_id from harmony.dev.ant_view_txs
where 1=1
and eth_to_address = '0xe53bf78f8b99b6d356f93f41afb9951168cca2c6'
and eth_from_address = :1
        )
, buy_sell_tx as (
select 
block_timestamp
,tx_id
,event_inputs:from as from_address
,event_inputs:to as to_address
,event_inputs:value as amount
,eth_contract_address
from harmony.dev.ant_logs
where 1=1
and tx_id in (select tx_id from dfk_item_buy_sell)
and event_emitted_name = 'Transfer'
)
, filtered_items as (
select 
m.block_timestamp
, m.tx_id
, m.from_address
, m.to_address
, m.amount / pow(10, t.decimals) as amount
, t.token_name
, t.token_symbol
, t.token_address
from buy_sell_tx m
LEFT JOIN harmony.dev.ant_tokens2 t on eth_contract_address = t.token_address
order by m.block_timestamp asc
)
select 
l.block_timestamp
,l.tx_id
,l.token_name as from_token
,l.amount as from_amount
,l.token_symbol as from_symbol
,r.token_name as to_token
,r.amount as to_amount
,r.token_symbol as to_symbol
,CASE
WHEN l.token_address='0x3a4edcf3312f44ef027acfd8c21382a5259936e7' then -l.amount * g.price
WHEN r.token_address='0x3a4edcf3312f44ef027acfd8c21382a5259936e7' then r.amount * g.price
ELSE 0
END as amount_usd
from filtered_items l
LEFT JOIN filtered_items r ON l.tx_id = r.tx_id and r.from_address = '0x0000000000000000000000000000000000000000'
LEFT JOIN harmony.dev.ant_view_token_price_dfkgold g on date(l.block_timestamp) = date(g.timestamp)
LEFT JOIN harmony.dev.ant_view_token_price_jewel p on date(l.block_timestamp) = date(p.timestamp)
where l.to_address = '0x0000000000000000000000000000000000000000'
order by l.block_timestamp asc
```

### PR DESCRIPTION
**Pull request by eevui**
# Description

Implements DFK item transactions table

# Tests 

- [x] Please provide evidence of your successful `dbt run` / `dbt test` here
```
root@396ff490794f:/harmony# dbt run -s dfk_item_transactions
Running with dbt=0.21.1
Found 42 models, 345 tests, 0 snapshots, 0 analyses, 180 macros, 0 operations, 8 seed files, 6 sources, 0 exposures

13:24:01 | Concurrency: 4 threads (target='dev')
13:24:01 |
13:24:01 | 1 of 1 START incremental model DEV.dfk_item_transactions............. [RUN]
13:24:58 | 1 of 1 OK created incremental model DEV.dfk_item_transactions........ [SUCCESS 1 in 57.15s]
13:24:58 |
13:24:58 | Finished running 1 incremental model in 65.03s.

Completed successfully

Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

# Checklist
- [x] Follow [dbt style guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md)
- [x] Tag the person(s) responsible for reviewing proposed changes @antonyip

### CODE REVIEW

File: models/dfk/dfk_item_transactions.sql [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        unique_key='tx_hash',
+        tags=['dfk'],
+        cluster_by=['block_timestamp']
+    )
+}}
+
+with incremental_txs as (
+    select *
+    from {{ ref("txs") }}
+    where {{ incremental_load_filter("ingest_at") }}
```
Comment (antonyip) [inline models/dfk/dfk_item_transactions.sql]:
```suggestion
    where {{ incremental_load_filter("ingested_at") }}
```

File: models/dfk/dfk_item_transactions.sql [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+),
+
+incremental_logs as (
+    select *
+    from {{ ref("logs") }}
+    where {{ incremental_load_filter("ingest_at") }}
```
Comment (antonyip) [inline models/dfk/dfk_item_transactions.sql]:
```suggestion
    where {{ incremental_load_filter("ingested_at") }}
```

File: models/dfk/dfk_item_transactions.sql [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        block_timestamp,
+        ingested_at,
+        native_from_address,
+        from_address,
+        case substring(market_txs.data, 1, 10)
+            when '0x9f37092a' then 'Buy'
+            when '0x096c5e1a' then 'Sell'
+        end as market_action,
+        to_number(gives.amount) as amount_give,
+        gives.token as token_give,
```
Comment (antonyip) [inline models/dfk/dfk_item_transactions.sql]: need to decimal this.. 
image

**Review by antonyip — CHANGES_REQUESTED** [review]
mostly good, some small fixes. 
Tested and checked.

File: models/dfk/dfk_item_transactions.sql (hunk 1 above)
Comment (eevui) [reply models/dfk/dfk_item_transactions.sql]: Oh, d'oh. Brainfart.

File: models/dfk/dfk_item_transactions.sql (hunk 3 above)
Comment (eevui) [reply models/dfk/dfk_item_transactions.sql]: Right, then might as well we add `token_name` and `token_symbol` as columns?

Also, how would I reconcile missing tokens from the `tokens` table? Do I leave them `NULL`?

**Comment by eevui** [comment]
Dropped and resynced `dfk_item_transactions` table.

File: models/dfk/dfk_item_transactions.sql (hunk 3 above)
Comment (antonyip) [reply models/dfk/dfk_item_transactions.sql]: leave them as null, we'll need to update when there are new tokens.

**Review by antonyip — APPROVED** [review]
Fixes Reviewed and Accepted.

**Pull request merged**